src/dune_rules/pkg_toolchain.mli

```diff
@@ -0,0 +1,28 @@
+open Import
+
+(** The path to the directory that will contain all toolchain
+    versions. Creates the directory if it doesn't already exist. *)
+val base_dir : unit -> Path.Outside_build_dir.t
+
+module Config : sig
+  (** Dune will download and build the ocaml-base-compiler and
+      ocaml-variants packages into a user-wide directory (shared among
+      projects) rather than using the usual package management mechanism to
+      install such packages. Currently compiler packages can't be installed
+      by dune's package management features as the compiler is not
+      relocatable, and this flag allows dune to workaround this problem
+      providing an experience to users that is almost identical to dune
+      installing the compiler packgae.
+
+      When this flag is disabled, users of dune package management need to
+      manage their compiler installation with opam or a system package
+      manager, as compilers packages that would be installed by dune will
+      not work correctly. *)
+  val enabled : bool Config.t
+
+  (** When building the compiler with toolchains, build it with [make
+      -j] rather than [make -j1], allowing more parallelism. This can
+      theoretically lead to build failures, but these are extremely rare in
+      practice. *)
+  val build_compiler_in_parallel : bool Config.t
```

Comment:
Why do we need this workaround for the compiler? If there are build failures, then they exist in regular opam as well. Or they also have an override?

Reply:
Prior to https://github.com/ocaml/dune/pull/10755, dune would always build all packages with `-j1`. The compiler takes particularly long to compile with `-j1`, and I've been told that building with `-j` might not work but I've never encountered any problems with it.

Reply:
Ack. Let's just follow what opam does and if this is indeed an issue, it's between opam repository and the compiler maintainers.